execution/block-partitioner/src/pre_partition/uniform_partitioner.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright © Aptos Foundation
+
+#[cfg(test)]
+use crate::test_utils::P2PBlockGenerator;
+use crate::{pre_partition::PrePartitioner, v2::types::PrePartitionedTxnIdx};
+use aptos_types::transaction::analyzed_transaction::AnalyzedTransaction;
+#[cfg(test)]
+use rand::thread_rng;
+
+/// Evenly divide txns. Example: processing txns 0..11 results in [[0,1,2,3],[4,5,6,7],[8,9,10]].
+pub struct UniformPartitioner {}
+
+impl PrePartitioner for UniformPartitioner {
+    fn pre_partition(
+        &self,
+        transactions: &[AnalyzedTransaction],
+        num_shards: usize,
+    ) -> Vec<Vec<PrePartitionedTxnIdx>> {
+        let num_items = transactions.len();
+        let num_chunks = num_shards;
+        let num_big_chunks = num_items % num_chunks;
```

Comment:
Calling it 'big_chunk' and 'small_chunk' will make reading the code harder. 
You can write this logic with 'chunk_size' instead of 'small_chunk_size'; 'num_chunks_with_overflow' or 'num_chunks_with_extra' instead of 'num_big_chunks'

Reply:
fixed

Reply:
cool, thanks